src/backend/distributed/commands/utility_hook.c

```diff
@@ -292,6 +294,19 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		PG_RE_THROW();
 	}
 	PG_END_TRY();
+
+	if (AddAllLocalTablesToMetadata && ShouldConvertTable && IsA(parsetree, CreateStmt) && IsCoordinator() && !IsBinaryUpgrade)
+	{
+		CreateStmt *createTableStmt = (CreateStmt *) parsetree;
+		if (createTableStmt->relation->relpersistence != 't')
```

Comment:
maybe also `createTableStmt->partbound == NULL` to avoid CREATE TABLE .. PARTITION OF issues (should be captured already by parent auto-converting.